sdk/spring/spring-cloud-azure-actuator/src/main/java/com/azure/spring/cloud/actuator/cosmos/CosmosHealthIndicator.java

```diff
@@ -45,19 +47,32 @@ public CosmosHealthIndicator(CosmosAsyncClient cosmosAsyncClient, String databas
 
     @Override
     protected void doHealthCheck(Builder builder) {
-        CosmosDatabaseResponse response = this.cosmosAsyncClient.getDatabase(database)
-            .read()
-            .block(timeout);
-
-        if (response != null) {
-            LOGGER.info("The health indicator cost {} RUs, cosmos uri: {}, dbName: {}",
-                response.getRequestCharge(), endpoint, database);
+        if (database == null) {
+            builder.status(Status.UNKNOWN).withDetail("Database not configured",
+                "The option of `spring.cloud.azure.cosmos.database` is not configured!");
+            return;
         }
-        if (response == null) {
-            builder.down();
-        } else {
-            builder.up().withDetail("database", response.getProperties().getId());
+
+        try {
+            CosmosDatabaseResponse response = this.cosmosAsyncClient.getDatabase(database)
+                                                                    .read()
+                                                                    .block(timeout);
+
+            if (response != null) {
+                LOGGER.info("The health indicator cost {} RUs, cosmos uri: {}, dbName: {}",
+                    response.getRequestCharge(), endpoint, database);
+                builder.up()
+                       .withDetail("RUs", response.getRequestCharge())
+                       .withDetail("CosmosUri", endpoint)
+                       .withDetail("database", database);
+            } else {
+                builder.down();
+            }
+        } catch (NotFoundException e) {
```

Comment:
I don't think we should catch `NotFoundException` here, the Javadoc of `certificateAsyncClient.getCertificateWithResponse("spring-cloud-azure-not-existing-certificate")` says `@throws ResourceNotFoundException when a certificate with {@code certificateName} doesn't exist in the key vault.`, but there's no such thing for the `cosmosAsyncClient.getDatabase(database).read()`, and `NotFoundException` is from the implementation package.